Generators/src/main/java/io/deephaven/pythonPreambles/KafkaToolsPreamble.txt

```diff
@@ -131,68 +134,69 @@ def consumeToTable(*args, **kwargs):
             ) from e
     elif not isinstance(offsets, jpy.JType):
         raise Exception(
-            "type " + type(offsets).__name__ +
-            "  of keyword argument 'offsets' not recognized; only str or dict allowed")
+            "value " + str(offsets) + " of type " + type(offsets).__name__ +
+            "  not recognized for argument 'offsets'; only str, dict or predefined constants allowed")
 
-    key = kwargs.pop('key', FROM_PROPERTIES)
-    value = kwargs.pop('value', FROM_PROPERTIES)
+    if key is None:
```

Comment:
It seems like you should use FROM_PROPERTIES in the method signature default.

Reply:
We should, but is sad and we can't.  FROM_PROPERTIES is an object that is not initialized by the time the def definition is processed.   If we say `def(..., some_arg=SOME_CONSTANT, ...)` for a `SOME_CONSTANT` that we initialize from a Java object static via jpy, when python processed the `def` it finds the initial value of `None` for the constant and default arguments are never evaluated again, only once at the time the def is processed, so it doesn't work.   See also https://illumon.slack.com/archives/C028TLCCSPR/p1629244827144200